modules/library/jdbc/src/main/java/org/geotools/data/jdbc/datasource/DBCPDataSourceFactory.java

```diff
@@ -39,8 +41,8 @@
  */
 public class DBCPDataSourceFactory extends AbstractDataSourceFactorySpi {
 
-    public static final Param DSTYPE = new Param("dstype", String.class,
-            "Must be DBCP", false);
+    public static final Param DSTYPE = new Param("dstype", String.class, "Must be DBCP", false,
+            null, Collections.singletonMap(Parameter.LEVEL, "program"));
```

Comment:
Not "DBTYPE", but seems to have the same purpose; presuming we should add this one too.